### PR TITLE
fix issue 859

### PR DESCRIPTION
**Pull request by dota17**
#859

### CODE REVIEW

File: include/yaml-cpp/node/convert.h [hunk 1]
```diff
@@ -133,6 +134,13 @@ inner_encode(const T& rhs, std::stringstream& stream){
       const std::string& input = node.Scalar();                            \
       std::stringstream stream(input);                                     \
       stream.unsetf(std::ios::dec);                                        \
+      if ((stream.peek() == '-') &&                                        \
+         (typeid(rhs) == typeid(unsigned) ||                               \
```
Comment (jbeder) [inline include/yaml-cpp/node/convert.h:138]: Prefer to specialize the template rather than check the typeid.

File: docs/Breaking-Changes.md [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
 ## HEAD ##
 
-_none_
+  * Add the exception throw when the negative is converted to unsigned integer for template `convert<type>`.
```
Comment (jbeder) [inline docs/Breaking-Changes.md:7]: Reword:

"Throws an exception when trying to parse a negative number as an unsigned integer."

File: docs/Breaking-Changes.md [hunk 3]
```diff
@@ -48,4 +48,4 @@ _none_
 
 ## 0.2.5 ##
 
-This wiki was started with v0.2.5.
\ No newline at end of file
+This wiki was started with v0.2.5.
```
Comment (jbeder) [inline docs/Breaking-Changes.md:51]: What change are you making here? Maybe an accidental newline?

File: docs/Breaking-Changes.md (hunk 2 above)
Comment (dota17) [reply docs/Breaking-Changes.md]: ok，done

File: docs/Breaking-Changes.md (hunk 3 above)
Comment (dota17) [reply docs/Breaking-Changes.md]: The vim editor automatically adds a newline symbol at the end of this file. I have removed it.

**Pull request merged**